### PR TITLE
Adding Kafka MultiDC source""

### PR DESCRIPTION
**Pull request by sajithshn**
Reverts wso2-extensions/siddhi-io-kafka#46

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/KafkaMultiDCSource.java [hunk 1]
```diff
@@ -18,8 +18,246 @@
...
+                        description = "This should contain the kafka server list which the kafka source should be "
+                                + "listening to. This should be given in comma separated values. "
+                                + "eg: 'localhost:9092,localhost:9093' ",
+                        type = {DataType.STRING}),
+                @Parameter(name = "topic",
+                        description = "The topic  which the source would be listening to. eg: 'topic_one' ",
+                        type = {DataType.STRING}),
+                @Parameter(name = "partition.no",
+                        description = "The partition number for the given topic",
+                        type = {DataType.STRING},
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/KafkaMultiDCSource.java:63]: Shall we make this as INT.

File: component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/SourceSynchronizer.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        toRemoveSeqNos.clear();
+    }
+
+    private synchronized void flushBuffer() {
+        logDebug("Start flushing buffer");
+        for (Map.Entry<Long, BufferValueHolder> entry : eventBuffer.entrySet()) {
+            Long sequenceNumber = entry.getKey();
+            BufferValueHolder eventHolder = entry.getValue();
+            if (sequenceNumber <= lastConsumedSeqNo) {
+                logDebug("Message with sequence " + sequenceNumber + " already received. " +
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/SourceSynchronizer.java:101]: Better have if debug is enabled before String concatenation.

File: component/src/test/java/org/wso2/extension/siddhi/io/kafka/multidc/KafkaMultiDCSourceSynchronizerTestCases.java [hunk 3]
```diff
@@ -0,0 +1,421 @@
+package org.wso2.extension.siddhi.io.kafka.multidc;
```
Comment (mohanvive) [inline component/src/test/java/org/wso2/extension/siddhi/io/kafka/multidc/KafkaMultiDCSourceSynchronizerTestCases.java:1]: License header is missing.

File: component/src/test/java/org/wso2/extension/siddhi/io/kafka/multidc/KafkaMultiDCSourceTestCases.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
+package org.wso2.extension.siddhi.io.kafka.multidc;
```
Comment (mohanvive) [inline component/src/test/java/org/wso2/extension/siddhi/io/kafka/multidc/KafkaMultiDCSourceTestCases.java:1]: License header is missing.

File: component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/KafkaMultiDCSource.java (hunk 1 above)
Comment (sajithshn) [reply component/src/main/java/org/wso2/extension/siddhi/io/kafka/multidc/source/KafkaMultiDCSource.java:63]: +1. This should be int

**Pull request merged**